### PR TITLE
Add PHP 8.4 support

### LINKED ISSUE #871: Support PHP 8.4
| Q | A
| --- | ---
| ParaTest version | 7.x
| PHPUnit version | unknown
| PHP version | 8.4.0-alpha3

#### Summary

Trying to install pest (see https://github.com/pestphp/pest/issues/1199) on php 8.4 results in a composer error.

#### Current behavior

Unable to install Pest on a newer version of PHP and see what is broken.

#### How to reproduce: command, code and error stack trace

`composer install`

#### Expected behavior

To be able to install on php 8.4

### PR DESCRIPTION
**Pull request by joanhey**
The tests pass, but show deprecations with the `symfony/string` v5.4.0 class in PHP 8.4 lowest test.

`Symfony/string` is fixed https://github.com/symfony/string/commit/bce75043af265dc8aca536a6ab1d6b3181763529 

~PD: perhaps it'll be better to change in `composer.json` :~
```
"require": {
        "php": "~8.2.0 || ~8.3.0 || ~8.4.0",
```
with
```
"require": {
        "php": ">=8.2",
```
 Changed to `"php": ">=8.2"`

Fix #871
that break PestPHP https://github.com/pestphp/pest/issues/1199

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -33,7 +33,7 @@
         }
     ],
     "require": {
-        "php": "~8.2.0 || ~8.3.0",
+        "php": ">=8.2",
```
Comment (Jean85) [inline composer.json]: Please do not put an open ended constraint here, the limited support to new PHP versions is intentional

**Review by Jean85 — CHANGES_REQUESTED** [review]

**Comment by Slamdunk** [comment]
I don't like the CI build on 8.4 to be successful: https://github.com/paratestphp/paratest/actions/runs/10473712966/job/29006479710?pr=872#step:3:19

`pcov` is not yet compatible with 8.4, https://github.com/krakjoe/pcov/pull/111

**Comment by Slamdunk** [comment]
Apparently there's an infinite loop somewhere with PHP 8.4, but I can't debug it yet locally because the pcov PECL extension hasn't been updated yet, we need to wait

**Comment by Slamdunk** [comment]
Superseded by https://github.com/paratestphp/paratest/pull/898 thank you anyway for your contribution

**Pull request closed without merging**